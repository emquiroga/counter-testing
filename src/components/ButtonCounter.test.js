const { render, screen } = require("@testing-library/react")
const { default: ButtonCounter } = require("./ButtonCounter")

describe('<ButtonCounter /> Tests', () => {
    const actionAdd = jest.fn()
    beforeEach(() => {
        render(<ButtonCounter value={3} name="add" action={actionAdd} />)
    })
    it('First snapshot', ()=> {
        expect(screen.getByLabelText("add")).toMatchSnapshot();
    })
    it('Creates a button properly? (Case add counter)', () => {
        expect(screen.getByLabelText("add")).toBeInTheDocument()
    })
    it('Creates a button properly without default value? (Case rest counter)', () => {
        render(<ButtonCounter name="rest" action={actionAdd} />)
        expect(screen.getByLabelText("rest")).toBeInTheDocument()
    })
})