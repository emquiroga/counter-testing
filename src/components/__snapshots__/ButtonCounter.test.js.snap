// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonCounter /> Tests First snapshot 1`] = `
<button
  aria-label="add"
>
  3
</button>
`;
